src/lazy/span.rs

```diff
@@ -63,6 +68,26 @@ impl<'a> Span<'a> {
     pub fn is_empty(&self) -> bool {
         self.bytes.is_empty()
     }
+
+    pub fn slice(&self, offset: usize, length: usize) -> Span<'a> {
```

Comment:
🪧 This PR modified the `TextBuffer` type to use a `Span` internally. A future PR will do the same for `BinaryBuffer`.